File: conformance/bundles/14-stateless-steps/squares.steps.ts
import { defineState } from '@oselvar/var'

// No state factory: these steps are pure — nothing to arrange, nothing to
// evolve — so defineState() is called bare and handlers get an empty state.
const { stimulus, sensor } = defineState()

stimulus('I warm up my mental math', () => {})

sensor('The square of {int} is {int}.', (_state, n: number) => [n, n * n])

File: conformance/bundles/14-stateless-steps/squares.steps.kt
@file:JvmName("SquaresSteps")

// Kotlin sibling of squares.steps.ts / squares.steps.py / SquaresSteps.java
// (bundle 14-stateless-steps): no state factory — these steps are pure, so
// defineState is called without one and handlers run against Unit.
package com.oselvar.varkt.conformance.bundle14

import com.oselvar.varkt.defineState
import com.oselvar.varkt.sensor

val steps = defineState {
    stimulus("I warm up my mental math") {}
    sensor("The square of {int} is {int}.") { n: Int -> listOf(n, n * n) }
}
